c/extras/ftrl.cc

```diff
@@ -165,28 +142,28 @@ dtptr Ftrl::test(const DataTable* dt) {
          j+= static_cast<size_t>(nth)) {
 
       hash_row(x, dt, j);
-      d_target[j] = predict(x, n_features + n_inter_features);
+      d_target[j] = predict_row(x, n_features + n_inter_features);
       if ((j+1) % REPORT_FREQUENCY == 0) {
         printf("Row: %zu\tPrediction: %f\n", j+1, d_target[j]);
       }
     }
   }
-
   return dt_target;
 }
 
 
 /*
 *  Make a prediction for an array of hashed features.
 */
-double Ftrl::predict(const Uint64Ptr& x, size_t x_size) {
+double FtrlModel::predict_row(const Uint64Ptr& x, size_t x_size) {
```

Comment:
I think your `x` is not the same as defined in the paper (McMahan&al. 2013). In the paper, `x` is the vector of features for the t-th observation. In your case, it's more like a sparse representation of `x`. I.e., your `x` contains indices of elements within the paper's `x` that are equal to 1, while all others are considered 0. Confusing...

Reply:
Yes, exactly, it is a sparse representation of `x`. Do you have a suggestion for a better name?